resources/definitions/fdmprinter.def.json

```diff
@@ -4088,6 +4088,31 @@
                     "settable_per_extruder": true,
                     "limit_to_extruder": "adhesion_extruder_nr"
                 },
+                "z_offset_layer_0":
+                {
+                    "label": "Initial Layer Z Offset",
+                    "description": "The extruder is offset from the normal height of the first layer by this amount. It can be positive (raised) or negative (lowered). Some filament types adhere to the build plate better if the extruder is raised slightly.",
+                    "unit": "mm",
+                    "type": "float",
+                    "default_value": 0,
+                    "minimum_value": "-layer_height_0 * 0.5",
+                    "minimum_value_warning": "-layer_height_0 * 0.2",
+                    "maximum_value_warning": "layer_height_0",
+                    "enabled": "resolveOrValue('adhesion_type') != 'raft'",
+                    "settable_per_mesh": false,
+                    "settable_per_extruder": false
+                },
+                "z_offset_taper_layers":
+                {
+                    "label": "Z Offset Taper Layers",
+                    "description": "When non-zero, the Z offset is reduced to 0 over that many layers. A value of 0 means that the Z offset remains constant for all the layers in the print.",
+                    "type": "int",
+                    "default_value": 0,
+                    "minimum_value": "0 if z_offset_layer_0 <= 0 or z_offset_taper_layers == 0 else max(1, round(5 * z_offset_layer_0 / layer_height + 0.5))",
```

Comment:
What's the reason for this minimum value of `5 * z_offset_layer_0 / layer_height + 0.5`? Unless it is physically impossible, I think just a minimum value of 0 would be fine.